### PR TITLE
Fix PCD load crashes on empty files

### PR DESCRIPTION
**Pull request by themightyoarfish**
addresses #5854

### CODE REVIEW

**Review by larshg — DISMISSED** [review]

File: common/include/pcl/common/io.h [hunk 1]
```diff
@@ -109,8 +109,15 @@ namespace pcl
   inline std::string
   getFieldsList (const pcl::PCLPointCloud2 &cloud)
   {
-    return std::accumulate(std::next (cloud.fields.begin ()), cloud.fields.end (), cloud.fields[0].name,
-        [](const auto& acc, const auto& field) { return acc + " " + field.name; });
+
```
Comment (themightyoarfish) [inline common/include/pcl/common/io.h:112]:
```suggestion
```

File: io/src/pcd_io.cpp [hunk 2]
```diff
@@ -397,7 +397,14 @@ pcl::PCDReader::readHeader (const std::string &file_name, pcl::PCLPointCloud2 &c
...
   {
-    PCL_ERROR ("[pcl::PCDReader::readHeader] Could not open file '%s'! Error : %s\n", file_name.c_str (), strerror (errno)); 
+    PCL_ERROR ("[pcl::PCDReader::readHeader] Could not open file '%s'! Error : %s\n", file_name.c_str (), strerror (errno));
+    fs.close ();
+    return (-1);
+  }
+
+  if (fs.peek() ==  std::ifstream::traits_type::eof())
+  {
+    PCL_ERROR ("[pcl::PCDReader::readHeader] File '%s' is empty.\n", file_name.c_str (), strerror (errno));
```
Comment (mvieth) [inline io/src/pcd_io.cpp:407]:
```suggestion
    PCL_ERROR ("[pcl::PCDReader::readHeader] File '%s' is empty.\n", file_name.c_str ());
```
The last part is unused, and I think `fs.peek()` wouldn't set `errno` anyway?

Comment (themightyoarfish) [reply io/src/pcd_io.cpp:407]: oops.

**Comment by themightyoarfish** [comment]
It appears that this change breaks some tests.

**Review by mvieth — APPROVED** [review]
Thanks!

> It appears that this change breaks some tests.

It was just a randomly failing test, not caused by your changes. I re-ran the test, now it succeeded.

**Review by larshg — APPROVED** [review]

**Pull request merged**